p2p_wallet/AppCoordinator/AppCoordinator.swift

```diff
@@ -0,0 +1,135 @@
+//
+//  AppCoordinator.swift
+//  p2p_wallet
+//
+//  Created by Chung Tran on 23/05/2022.
+//
+
+import AnalyticsManager
+import Foundation
+import Resolver
+import SolanaSwift
+import UIKit
+
+class AppCoordinator {
+    // MARK: - Dependencies
+
+    private var appEventHandler: AppEventHandlerType = Resolver.resolve()
+    private let storage: AccountStorageType & PincodeStorageType & NameStorageType = Resolver.resolve()
+    let analyticsManager: AnalyticsManager = Resolver.resolve()
+    let notificationsService: NotificationService = Resolver.resolve()
+
+    // MARK: - Properties
+
+    let window: UIWindow
+    let lockView = LockView()
+    let indicator = WLLoadingIndicatorView(isBlocking: true)
+    var isRestoration = false
+    var showAuthenticationOnMainOnAppear = true
+    var resolvedName: String?
+
+    // MARK: - Initializers
+
+    init(window: UIWindow) {
+        self.window = window
+        defer { appEventHandler.delegate = self }
+    }
+
+    // MARK: - Methods
+
+    func start() {
+        Task { await reload() }
+    }
+
+    func reload() async {
+        // remove rootvc, show loading
+        await MainActor.run {
+            window.rootViewController = nil
+            // add lockview
+            lockView.removeFromSuperview()
+            window.addSubview(lockView)
+            lockView.autoPinEdgesToSuperviewEdges()
+
+            // show loading
+            indicator.removeFromSuperview()
+            window.addSubview(indicator)
+            indicator.autoPinEdgesToSuperviewEdges()
```

Comment:
will it be better to extract this piece of `MainActor.run` code to a separate function? to simplify this method

Reply:
yes of course. I'll do it